packages/components/text-link/src/TextLink.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { HTMLProps } from 'react';
+import { cx } from 'emotion';
+import { Flex, CommonProps } from '@contentful/f36-core';
+import { styles } from './TextLink.styles';
+import { TextLinkVariant } from './types';
+import { Icon, IconComponent } from '@contentful/f36-icon';
+
+export interface TextLinkProps
+  extends Omit<
```

Comment:
I think we should use Polymorphic here to create proper typings based on `as` attribute passed here.

Reply:
What use cases do you see for `as` prop, that couldn't be covered with the current implementation? By current implementation I mean:
- pass href -> get `a`
- don't pass href -> get `button`
BTW the same approach used in the new Button component. For me there is no difference between the `Button` and `TextLink` components, only styling.
What do you think?

Reply:
I think it's a matter of consistency, all other basic components are using `as` attributes. A very cool feature of it that once you declare `as="a"` it automatically understands that `href`, `rel` and other specific attributes are possible, if it's button then we have attribute types for button.

If the new Button uses the same approach then we need to change that.